typing/typecore.ml

```diff
@@ -5532,6 +5768,25 @@ let report_type_expected_explanation expl ppf =
   | When_guard ->
       because "a when-guard"
 
+let escaping_hint reason context =
+  match reason, context with
+  | Types.Value_mode.Locality, Some Return ->
+      [ Location.msg
+          "@[Hint: Cannot return local value without an explicit@ \
+           \"local_\" annotation@]" ]
+  | Types.Value_mode.Locality, Some Tailcall_argument ->
+      [ Location.msg
+          "@[Hint: Local values cannot be arguments to tail-calls@ \
+           unless the result of the tail-call has a \"local_\" annotation@]" ]
```

Comment:
I think we should reword this hint. From some experimentation, this is an extremely common issue. This hint, while accurate, suggests that the fix is to add a `local_` annotation to the result of the called function. This is almost never the right thing to do: usually, you should make the call non-tail instead.

Suggestion:
```
This argument cannot be local, because this call is a tail call
```
(I wouldn't even bother mentioning the possibility of making the result `local_`: it's certainly a thing you can do, but it's unlikely to help someone who sees this error, so I wouldn't include it in the hint)